tests/io_adapters_test.py

```diff
@@ -0,0 +1,77 @@
+import tensorflow as tf
+from tensorflow import keras
+import numpy as np
+import unittest
+
+from lib import io_adapter as adp
+
+
+
+class InputAdapterTest(tf.test.TestCase):
+  def setUp(self):
+    super(InputAdapterTest, self).setUp()
+    self.default_size = 32
+    self.default_depth = 64
+
+  # tests if the output of the adaptes is of correct shape
+  def test_output_shape(self):
+    input_shape = (64, 64, 3)
+    batch_size = 16
+    expected_out_shape = (batch_size, self.default_size,
+                          self.default_size, self.default_depth)
+    adapter = self._create_default_adapter(input_shape)
+    out = adapter(np.zeros((batch_size, *input_shape)))
+    self.assertShapeEqual(np.zeros(expected_out_shape), out)
+
+  def test_small_out_size(self):
```

Comment:
These are usually self-explanatory, but you can also add short docstring where it may be clarifying things.